### PR TITLE
Remove gated content feature flags from mobile

### PR DESCRIPTION
**Pull request by dharit-tan**
### Description
Removes all usages of `useIsGatedContentEnabled`, `useIsSpecialAccessEnabled`, `useIsCollectibleGatedEnabled` from mobile.

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

Local ios stage - tested track tiles, lineup tiles, upload flow

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/screens/edit-track-screen/screens/RemixSettingsScreen.tsx [hunk 1]
```diff
@@ -235,11 +233,7 @@ export const RemixSettingsScreen = () => {
             />
           ) : null}
           <View style={styles.option}>
-            <Text {...labelProps}>
-              {isGatedContentEnabled
-                ? messages.markRemix
-                : messages.isRemixLabel}
```
Comment (piazzatron) [inline packages/mobile/src/screens/edit-track-screen/screens/RemixSettingsScreen.tsx:241]: can we lose `isRemixLabel`?

File: packages/mobile/src/screens/edit-track-screen/screens/RemixSettingsScreen.tsx [hunk 2]
```diff
@@ -255,11 +249,7 @@ export const RemixSettingsScreen = () => {
                 styles={{ root: styles.inputRoot, input: styles.input }}
                 value={remixOfInput}
                 onChangeText={handleChangeLink}
-                placeholder={
-                  isGatedContentEnabled
-                    ? messages.enterLink
-                    : messages.remixUrlPlaceholder
```
Comment (piazzatron) [inline packages/mobile/src/screens/edit-track-screen/screens/RemixSettingsScreen.tsx:261]: Same for these

File: packages/mobile/src/screens/edit-track-screen/screens/TrackAvailabilityScreen.tsx [hunk 3]
```diff
@@ -147,20 +142,16 @@ export const TrackAvailabilityScreen = () => {
...
+      value: specialAccessAvailability,
+      disabled: noSpecialAccess
+    },
+    {
+      label: collectibleGatedAvailability,
+      value: collectibleGatedAvailability,
+      disabled: noCollectibleGate
+    },
     { label: hiddenAvailability, value: hiddenAvailability, disabled: noHidden }
   ].filter(removeNullable)
```
Comment (piazzatron) [inline packages/mobile/src/screens/edit-track-screen/screens/TrackAvailabilityScreen.tsx:156]: can lose this removeNullable now

File: packages/mobile/src/screens/edit-track-screen/fields/TrackVisibilityField.tsx [hunk 4]
```diff
@@ -68,18 +67,10 @@ export const TrackVisibilityField = (props: TrackVisibilityFieldProps) => {
     return [messages.public]
   }, [premiumConditions, isUnlisted, fieldVisibilityLabels])
 
-  const isGatedContentEnabled = useIsGatedContentEnabled()
-  const label = isGatedContentEnabled
-    ? messages.availability
-    : messages.trackVisibility
```
Comment (piazzatron) [inline packages/mobile/src/screens/edit-track-screen/fields/TrackVisibilityField.tsx:74]: Can we lose these old messages?

**Review by piazzatron — APPROVED** [review]
Thanks for taking on this cleanup!! Looks great

**Comment by piazzatron** [comment]
I think name of PR is wrong though :)

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rt-flags

**Review by sddioulde — COMMENTED** [review]
Is this taking into account that we still want the feature flags on iOS?

also nit: `useIsGatedContentEnabled` is twice on the PR description

**Comment by dharit-tan** [comment]
> Is this taking into account that we still want the feature flags on iOS?
> 
> also nit: `useIsGatedContentEnabled` is twice on the PR description

Turns out we only use those in the upload flow, don't filter out tracks in lineup. Had a discussion with @piazzatron on slack and we decided to just delete, since they don't really do what we want anyway.

**Pull request merged**